libmscore/edit.cpp

```diff
@@ -1053,9 +1053,26 @@ void Score::regroupNotesAndRests(const Fraction& startTick, const Fraction& endT
                                     continue; // this voice is empty here
                               if (!cr->isRest() || s->tick() + cr->actualTicks() > maxTick || toRest(cr)->isGap())
                                     break; // next element in the same voice is not a rest, or it exceeds the selection, or it is a gap
+                              if (cr->tuplet()) {
+                                    Tuplet* tuplet = cr->tuplet(); // element is a tuplet; special case
+                                    short tupletSize = tuplet->elements().size();
+                                    bool tupletAllRests = true;
+                                    for (auto const& el : tuplet->elements()) {
+                                          // Check all tuplet elements
+                                          if (!el->isRest())
+                                                tupletAllRests = false; // Tuplet element is not a rest
+                                        }
+                                    if (!tupletAllRests) break;
+                                    // Set to last element (i.e. rest) of the tuplet
+                                    cr = toRest(tuplet->elements().back());
+                                    for (int i = 0; i < tupletSize - 1; i++) {
```

Comment:
Hmm, why `tupletSize - 1` instead of `tupletSize`?

Reply:
Because the outer for loop will already call next after, so when using `tupletSize`, the next note will be skipped

Reply:
Oh right, didn't think of that